pkg/controller/operators/catalog/operator.go

```diff
@@ -346,7 +346,7 @@ func NewOperator(ctx context.Context, kubeconfigPath string, clock utilclock.Clo
 		subscription.WithAppendedReconcilers(subscription.ReconcilerFromLegacySyncHandler(op.syncSubscriptions, nil)),
 		subscription.WithRegistryReconcilerFactory(op.reconciler),
 		subscription.WithGlobalCatalogNamespace(op.namespace),
-		subscription.WithSourceProvider(resolverSourceProvider),
+		subscription.WithSourceProvider(op.sourceInvalidator),
```

Comment:
So just to clarify, we needed the list of catalogSources in https://github.com/operator-framework/operator-lifecycle-manager/commit/b9638dcaaa14d4005b12abe63feb2055a49ca793 and we just used the wrong source provider?

Reply:
Yeah that's right, the `OperatorGroupToggleSourceProvider` (from what I understand) is essentially a filter which allows users to exclude global catalogs from resolution. But I don't believe that this `sourceProvider` was intended to be used when taking cache snapshots of the `CatalogSource` namespace, instead it's used to allow for instance a user to ignore a Catalog update by adding an annotation to the `OperatorGroup` of a separate namespace in which their operator is installed.

In OCP deployments, the `openshift-marketplace` namespace doesn't have an `OperatorGroup`, so if you use this source provider it will error out when taking snapshots and give you nothing back.

Reply:
Ah got it, thanks Daniel!